FirebaseSessions/Sources/Time.swift

```diff
@@ -0,0 +1,26 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+protocol TimeProvider {
```

Comment:
Since this is a "provider" pattern only used by session start event, could you take a look at https://github.com/firebase/firebase-ios-sdk/pull/10360 and see if, in this case, would it make sense to inject function instead of object?

Reply:
That's a good question - I've been favoring the Provider pattern for 2 reasons:
 - Consistency
 - If there's another class that needs Time, then it's easy to move it around.

In this case I don't see another use case for time, so I'm down to go with the function approach.

Reply:
Ehh actually I might keep the Provider. After writing up the functional solution it seems less readable to me